### ISSUE TITLE
Create terminal-core library for use in Code terminal plugin

### ISSUE THREAD

**Issue by jeremypw** [elementary/terminal#481]
I have searched open and closed issues for duplicates.

## Feature
The terminal plugin in Code does not have the same behaviour and features as the elementary Terminal app.  For consistency it would be good if the same Terminal Widget and functionality could be used in both app.

**Describe the solution you'd like**
Create a Terminal core library containing the Terminal Widget and other necessary code which can be used be the Code terminal plugin and provide the same functionality.

**Describe alternatives you've considered**
The features desired could be provided by copying Terminal code into the Code terminal plugin.

**Additional context**
See https://github.com/elementary/code/issues/155 and

**Comment by danrabbit** [comment]
So I guess something to think about is what do we actually expect the Terminal to be used for and why does it exist here?

If the idea is having access to a full Terminal, I think we should probably say that it would be better to tile the actual Terminal app.

The only thing I've ever used it for is to build and run Terminal. I'd be interested to hear from other people what they're using this plugin for.

My guess is that it would be better time spent making a build & run feature than trying to embed the whole of Terminal

@cassidyjames thoughts?

**Comment by jeremypw** [comment]
I agree this is not something to attempt without real need in Code.  It may be better to manage user expectations of the Code terminal plugin. I too have mainly used the Code terminal when testing Terminal.  The only thing is, if there *is* to be a terminal in Code it could confuse users if the keyboard behaviour is not the same.  I wouldn't necessaryily expect it to have the search feature for example.

**Comment by danrabbit** [comment]
Yeah my guess is that it might be better to completely retire the terminal plugin in favor of a tighter scoped build and run
